### ISSUE TITLE
Numbers Spelling

### ISSUE THREAD

**Issue by MobinaPoulaei** [k2-fsa/OmniVoice#174]
### Checks

- [x] This template is only for research question, not usage problems, feature requests or bug reports.
- [x] I have thoroughly reviewed the project documentation and read the related paper(s).
- [x] I have searched for existing issues, including closed ones, no similar questions.
- [x] I am using English to submit this issue to facilitate community communication.

### Question details

Hi,

I tested the model with the sentence:

"I have 2345 apples."

The generated speech did not read the number correctly. It seems that the model does not properly handle numbers larger than 1000. Is this a known limitation, and is there a recommended way to address it?

Thanks.

**Comment by CHRONOMASTER** [comment]
Try replacing all multi digit numbers with the word equivalent.
For example, instead of 2345, replace it with two thousand three hundred forty-five.

**Comment by MobinaPoulaei** [comment]
> Try replacing all multi digit numbers with the word equivalent. For example, instead of 2345, replace it with two thousand three hundred forty-five.

Yes, I know that would work, but I was curious whether there is a better approach than manually normalizing numbers before inference.

**Comment by celguar** [comment]
There is - let library like num2words automatically normalize numbers before inference

```
def convert_numbers_to_words(text):
    from num2words import num2words
    """Convert all numbers in text to their word representation"""

    def replace_number(match):
        num = match.group()
        words = num2words(num)
        # Remove all commas
        words = words.replace(',', '')
        return words

    # Find all numbers in text
    pattern = r'\b\d+\b'
    return re.sub(pattern, replace_number, text)
```

**Comment by zhu-han** [comment]
Text normalization is the solution. We mentioned this in README.md: `For better results with Arabic numerals, normalize them to words first (e.g., "123" → "one hundred twenty-three") with text normalization tools (e.g., [WeTextProcessing](https://github.com/wenet-e2e/WeTextProcessing)).`

**Comment by programmeddeath1** [comment]
Hii Can additional training data help fix the spoken number issue? especially if I want to train it on other languages? If I add more data with those number pronunciations in that specific language, will it improve the models performance on them. If yes, what would be the number of samples we would need to finetune the model in for the same?

**Comment by celguar** [comment]
what's wrong with num2words? It supports many languages
